src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -658,18 +654,27 @@ private void processLogsAndUpload() throws InterruptedException {
                     Collectors.toList());
 
             if (unitsOfWork.isEmpty()) {
-                TimeUnit.SECONDS.sleep(periodicUpdateIntervalSec);
                 isCurrentlyUploading.set(false);
             } else {
                 unitsOfWork.forEach((unit) -> {
                     CloudWatchAttempt cloudWatchAttempt = logsProcessor.processLogFiles(unit);
                     uploader.upload(cloudWatchAttempt, 1);
                 });
+                emitEventStatus(EventType.LOG_GROUP_PROCESSED);
```

Comment:
Maybe time to change the event name given each cycle might process multiple log groups on the `while` loop